Add CPack distribution

antoniovazquezblanco opened the pull request:
- An action wich calls cmake packaging target and uploads all platform corresponding artifacts.
- Removed all packaging related commands in the main pipeline.
- Added CPack distribution with support for DEB, RPM and TGZ binary distribution in linux, ZIP distribution in windows and the default distribution on Mac (I have no means to test this...)
- Removed previous DEB packaging script

antoniovazquezblanco commented:
Superseeds https://github.com/BatchDrake/sigutils/pull/34 I believe...

BatchDrake commented on src/CMakeLists.txt:
```diff
@@ -113,3 +113,13 @@ target_pc_file_generate(sigutils "Digital signal processing utility library")
 install(FILES ${PROJECT_BINARY_DIR}/src/sigutils.pc DESTINATION ${CMAKE_INSTALL_PKGCONFIGDIR})
+
+# General packaging settings
+set(CPACK_PACKAGE_NAME Sigutils)
```
Will this create a Debian package with that capital S in the name?

antoniovazquezblanco replied:
Yep, I can fix that.

antoniovazquezblanco commented:
TODO: Verify that generated packages have dependency info.

antoniovazquezblanco commented:
Removed RPM generation as it was not tested.

Fixed debian generation in order to produce two packages one lib and one devel with headers and pc file. Also set the deps in the packages.

I hope this looks better now.
